Editors/RoomEditor.ui

```diff
@@ -17,7 +17,1381 @@
+   </property>
+   <item>
+    <layout class="QHBoxLayout" name="contentLayout" stretch="0,1">
+     <property name="spacing">
+      <number>4</number>
+     </property>
+     <item>
+      <widget class="QTabWidget" name="propertiesTabWidget">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
+         <horstretch>0</horstretch>
+         <verstretch>0</verstretch>
+        </sizepolicy>
+       </property>
+       <property name="currentIndex">
+        <number>2</number>
+       </property>
+       <property name="elideMode">
+        <enum>Qt::ElideNone</enum>
+       </property>
+       <property name="usesScrollButtons">
+        <bool>false</bool>
+       </property>
+       <property name="documentMode">
+        <bool>false</bool>
+       </property>
+       <property name="tabBarAutoHide">
+        <bool>false</bool>
+       </property>
+       <widget class="QWidget" name="layersTab">
+        <attribute name="title">
+         <string>Layers</string>
+        </attribute>
+        <layout class="QVBoxLayout" name="verticalLayout_6">
+         <property name="spacing">
+          <number>4</number>
+         </property>
+         <property name="leftMargin">
+          <number>4</number>
+         </property>
+         <property name="topMargin">
+          <number>4</number>
+         </property>
+         <property name="rightMargin">
+          <number>4</number>
+         </property>
+         <property name="bottomMargin">
+          <number>4</number>
+         </property>
+         <item>
+          <widget class="QTreeWidget" name="layersTreeWidget">
+           <property name="sizePolicy">
+            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
+             <horstretch>0</horstretch>
+             <verstretch>1</verstretch>
+            </sizepolicy>
+           </property>
+           <attribute name="headerVisible">
+            <bool>false</bool>
+           </attribute>
+           <column>
+            <property name="text">
+             <string>Name</string>
+            </property>
+           </column>
+           <item>
+            <property name="text">
+             <string>Background 0</string>
+            </property>
+           </item>
+           <item>
+            <property name="text">
+             <string>Background 1</string>
+            </property>
+           </item>
+           <item>
+            <property name="text">
+             <string>NPC</string>
+            </property>
+            <item>
+             <property name="text">
+              <string>Enemy 1</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>Enemy 2</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>New Item</string>
+             </property>
+            </item>
+            <item>

```

Comment:
This generator fucking sucks. Just saying. This is atrocious, and there's no reason for it.

Reply:
Agreed, QML is a lot cleaner, but that requires adding too much more Qt shit onto this. Don't get me wrong, I think QML is really nice, but it requires a whole 'nother engine on top of the other Qt stuff just to render it. Our hands are pretty much tied on this issue, and it doesn't bother me anyway because most of these generators are garbage. Also, I'm just glad this garbage isn't stuffed into the classes like it was in LGM, which makes the actual logic really hard to read.